_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,374 @@
+
+- You can increase the number of partitions / replicas which can help with increasing the throughput of topics.
+
+## Partition Reassignment Throttle
+Reassigning partitions between brokers often leads to additional interbroker network traffic, in addition to the normal traffic required for replication. To avoid overloading the cluster, it is recommended to always set a throttle rate to limit the bandwidth used by the reassignment. 
+This can be done using the `--throttle` flag which sets the maximum allowed bandwidth in bytes per second, for example `--throttle 5000000` sets the limit to 5 MB/s.
+
+Throttling might cause the reassignment to take longer to complete.
+
+1. If the throttle is too low, the newly assigned brokers will not be able to keep up with records being published and the reassignment will never complete.
+
+2. If the throttle is too high, the overall health of the cluster may be impacted.
+
+You can use the `kafka.server:type=FetcherLagMetrics,name=ConsumerLag,clientId=([-.\w]+),topic=([-.\w]+),partition=([0-9]+)` metric (which with the spec.kafka.metrics settings in your Kafka CR would appear be scraped by Prometheus as `kafka_server-fetcherlagmetrics_consumerlag`) to observe how far the followers are lagging behind the leader. You can also refer to this [example](https://github.com/strimzi/strimzi-kafka-operator/blob/main/packaging/examples/metrics/kafka-metrics.yaml) for configuring metrics 
+
+The best way to set the value for throttle is to start with a safe value.
+If the lag is growing, or decreasing too slowly to catch up within a reasonable time, then the throttle should be increased.
+To do this, run the command again to increase the throttle, iterating until it looks like the broker will join the ISR within a reasonable time.
+
+## Actions that can be executed while using the tool
+
+It has three different actions:
+
+| Action | Description |
+| :-----------: | ------------- |
+| `generate`  | Takes a set of topics and brokers and generates a reassignment JSON file. This action is optional if we already have `reassignment.json` file with us. You can use this action using `--generate`|
+| `execute` | Takes a reassignment JSON file and applies it to the partitions and brokers in the cluster. The `reassignment.json` file can be either the one proposed by the `generate` action or written by the user. `--execute`  is used to carry out this action |
+| `verify`  | Using the same reassignment JSON file as the `--execute` step, `--verify` checks whether all the partitions in the file have been moved to their intended brokers. If the reassignment is complete, `--verify` also removes any replication quotas (`--throttle`) that are in effect. Unless removed, throttles will continue to affect the cluster even after the reassignment has finished. This action can be executed using `--verify` |
+
+## Example time
+
+Let us understand the working of this tool with a concrete example.
```

Comment:
Why is this example easier than using Cruise Control and `remove-brokers` rebalance? If you want to use this as an example, you probably need to set it into a more context, link to the way to do it with Cruise Control etc.